### PR TITLE
Return miekg/dns Errors to Practitioners

### PR DESCRIPTION
**Pull request by bendbennett**
Closes: #227

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -619,7 +619,11 @@ func resourceDnsDelete(d *schema.ResourceData, meta interface{}, rrType uint16)
 
 		msg.SetUpdate(d.Get("zone").(string))
 
-		rr, _ := dns.NewRR(fmt.Sprintf("%s 0 %s", fqdn, dns.TypeToString[rrType]))
+		rr, err := dns.NewRR(fmt.Sprintf("%s 0 %s", fqdn, dns.TypeToString[rrType]))
+		if err != nil {
+			return fmt.Errorf("error reading DNS record: %s", err)
```
Comment (bflad) [inline internal/provider/provider.go:624]: Should we include the DNS record string in the error message or is it already there?

**Review by bflad — APPROVED** [review]
Looks good to me 🚀 (If the RR is not present in the error message, it might be good to include it for troubleshooting purposes 👍 )

File: internal/provider/provider.go (hunk 1 above)
Comment (bendbennett) [reply internal/provider/provider.go:624]: It doesn't look to me that the error message contains the DNS record string.

In which case, are you referring to doing a nil check on `rr` and then calling `rr.String()` and adding this to the error message if it is available?

Comment (bflad) [reply internal/provider/provider.go:624]: I was thinking the available `fmt.Sprintf("%s 0 %s", fqdn, dns.TypeToString[rrType])` content

**Pull request merged**